Develop

yonatandt opened the pull request.

yonatandt committed:
feat(users): adds the following routes:
- get user by ID.
- get users by partial name.

Signed-off-by: YonatanTal <yonatald@gmail.com>

yonatandt committed:
fix: minor fixes

Signed-off-by: YonatanTal <yonatald@gmail.com>

vitaly-furman committed:
Merge pull request #1 from AlphaWolfTeam/feature/user

WIP: basic user routes

lustigy96 commented on src/router.ts:
```diff
@@ -0,0 +1,26 @@
+import { InvalidArgument } from './utils/errors/client.error';
+import UserController from './user/user.controller';
+import { wrapAsync } from './utils/wrappers';
+
```
the router role is to pass the request to the controller. Only the controller suppose to send the responses like (res.status, res.send)

lustigy96 commented on src/router.ts:
```diff
@@ -0,0 +1,26 @@
+  const partialName = req.query.partialName;
+  if (!partialName) {
+    throw new InvalidArgument("argument 'partialName' was not sent in the query params");
+  }
```
the router should not emit. This is the manager job. 
router: pass requests 
contoller: responses
manager: analyze the request

lustigy96 commented on src/user/user.interface.ts:
```diff
@@ -0,0 +1,60 @@
+  clearance?: string;
+  // Calculated
+  fullName?: string;
+}
```
Here you and @carmeltsadok should compare the IUser. For whatever reason she gets another structure.

carmeltsadok commented on src/router.ts:
```diff
@@ -0,0 +1,26 @@
+  res.status(200).send('alive');
+});
+
+appRouter.get('/api/users/:id', wrapAsync(async (req, res) => {
```
I think that the api-gateway should direct the routes to users, groups, etc. (I also think that because the group-service's routes don't start like that it should match). I would change the routes to '/:id' and '/'.

carmeltsadok commented on src/router.ts:
```diff
@@ -0,0 +1,26 @@
+  res.send(user);
+}));
+
+appRouter.get('/api/users', async (req, res) => {
```
Why the function is not wrapped by wrapAsync?

carmeltsadok commented on src/router.ts:
```diff
@@ -0,0 +1,26 @@
+import UserController from './user/user.controller';
+import { wrapAsync } from './utils/wrappers';
+
+const appRouter = Router();
```
In my opinion, the router file should be cleaner. The content of each wrapped function should be in a separate file, in the controller.

yonatandt replied:
This is a very lightweight service with a meager amount of logic, most of the code just handles errors. Therefore I see no need to separate the route handling to lots of middlewares and and files, like a large service usually has.
I can see why you want the routes to look cleaner, I'll try to make it neater without creating more redundant files.

carmeltsadok commented on src/user/user.controller.ts:
```diff
@@ -0,0 +1,55 @@
+import axios, { AxiosResponse } from 'axios';
+import { IUser } from './user.interface';
+
+export default class UserController {
```
In my opinion, you should add a manager that does the same as those functions.

yonatandt committed:
minor fixes

Signed-off-by: YonatanTal <yonatald@gmail.com>

yonatandt committed:
Merge pull request #4 from AlphaWolfTeam/feature/user

Feature/user

yonatandt committed:
refactor:
- minimizes docker image size
- orginizes the code more neatly between the files

Signed-off-by: YonatanTal <yonatald@gmail.com>

yonatandt committed:
fix

Signed-off-by: YonatanTal <yonatald@gmail.com>

yonatandt committed:
fix(security): fixes 2 high severity vulnerabilities

Signed-off-by: YonatanTal <yonatald@gmail.com>

yonatandt committed:
Merge pull request #6 from AlphaWolfTeam/feature/user

lustigy96 committed:
Feat/hierarchy flat (#8)

* feat: add the flatHierarchy feature
#closes #5 

Co-authored-by: Yonatan Tal <yonatald@gmail.com>

carmeltsadok committed:
Create develop.actions.yml

carmeltsadok committed:
Create version-tag.yml

carmeltsadok committed:
Merge pull request #11 from alphawolfteam/ci

Create develop.actions.yml

yonatandt committed:
feat(apm): add apm transactions on requests. (#10)

* feat(apm): add apm transactions on requests.

Signed-off-by: YonatanTal <yonatald@gmail.com>

carmeltsadok committed:
Update develop.actions.yml

Add update azure vm step

carmeltsadok committed:
Merge pull request #12 from alphawolfteam/ci

Update develop.actions.yml
